Lib/pathlib.py

```diff
@@ -1265,19 +1265,33 @@ def lstat(self):
     def rename(self, target):
         """
         Rename this path to the given path.
+
+        If target is an instance of Path returns it, otherwise creates
+        new instance from target (string) and returns it.
         """
         if self._closed:
             self._raise_closed()
         self._accessor.rename(self, target)
+        if isinstance(target, str):
+            target = self.__class__(target)
```

Comment:
It might be safer to call `self.__class__` unconditionally (i.e. if a new type of object besides `PurePath` or `str` is added to `pathlib` in the future).  If we want to support original object return in `pathlib`, that's really something that should be added to `PurePath._from_parts`.

```python
if self._closed:
    self._raise_closed()
self._accessor.rename(self, target)
return self.__class__(target)
```

Reply:
Yes, I was considering this approach as well, just wanted to avoid having two objects representing the same thing.
I have no strong preferences for either approach